lib/snippet.js

```diff
@@ -0,0 +1,63 @@
+import { IdentityVerification } from './index';
+
+export default class Snippet {
+  constructor(settings) {
+    if (!settings.app_id) {
+      throw new Error('You must provide an app_id in your Intercom settings');
+    }
+    if (!settings.verificationSecret) {
+      throw new Error('You must provide your verification secret in your Intercom settings');
+    }
+    if (!settings.user_id && !settings.email) {
+      throw new Error('You must provide email or user_id in your Intercom settings');
+    }
+    this.settings = settings;
+  }
+  create() {
+    const verificationSecret = this.getVerificationSecret();
+    const identifier = this.getIdentifier();
+    const userHash = IdentityVerification.userHash({
+      secretKey: verificationSecret,
+      identifier
+    });
+    this.settings.user_hash = userHash;
+    return this.generateSnippetHTML();
+  }
+  getVerificationSecret() {
+    const { verificationSecret } = this.settings;
+    delete this.settings.verificationSecret;
+    return verificationSecret;
+  }
+  getIdentifier() {
+    if (this.settings.user_id) {
+      return this.settings.user_id.toString();
+    } else {
+      return this.settings.email;
+    }
+  }
+  generateSnippetHTML() {
+    return `
+<script>
+  window.intercomSettings = {
+    ${this.settingsToString(this.settings)}
+  };
+</script>
+<script>(function(){var w=window;var ic=w.Intercom;if(typeof ic==="function"){ic('reattach_activator');ic('update',intercomSettings);}else{var d=document;var i=function(){i.c(arguments)};i.q=[];i.c=function(args){i.q.push(args)};w.Intercom=i;function l(){var s=d.createElement('script');s.type='text/javascript';s.async=true;s.src='https://widget.intercom.io/widget/${this.settings.app_id}';var x=d.getElementsByTagName('script')[0];x.parentNode.insertBefore(s,x);}if(w.attachEvent){w.attachEvent('onload',l);}else{w.addEventListener('load',l,false);}}})()</script>
+    `;
+  }
+  settingsToString(settings) {
+    const intercomSettings = [];
+    Object.keys(settings).map(key => {
+      if (typeof settings[key] === 'object' && settings[key] !== null) {
+        intercomSettings.push(`${key}: { ${this.settingsToString(settings[key])} }`);
+      } else {
+        if (typeof settings[key] === 'string') {
+          intercomSettings.push(`${key}: "${settings[key]}"`);
+        } else {
+          intercomSettings.push(`${key}: ${settings[key]}`);
+        }
+      }
+    });
+    return intercomSettings.join(', ');
```

Comment:
`intercomSettings` needs to be an array?

Reply:
Doesn't have to be. I figured keeping each key / value pair as an element in an array makes it much easier to join them into a string and not have to worry about where you need to add commas.

Reply:
👍

Reply:
[JSON.stringify](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) won't work here?

Reply:
No, JSON.stringify will quote literally everything (including the braces for the hash), so you'd get this:

```javascript
const settings = { 
  app_id: 'abc123', 
  name: 'Peter McKenna', 
  user_id: 1 
};

// Returns "{"app_id":"abc123","name":"Peter McKenna","user_id":1}"
JSON.stringify(settings);
```